src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -297,27 +298,124 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
 
-            // The symbols may include local variables that are declared later in the method and
-            // should not be included in the completion list, so remove those. Filter them away,
-            // unless we're in the debugger, where we show all locals in scope.
-            if (filterOutOfScopeLocals)
-                symbols = symbols.WhereAsArray(symbol => !symbol.IsInaccessibleLocal(_context.Position));
+                if (symbol.IsExtensionMethod() &&
+                    !Equals(enclosingNamedType, symbol.ContainingType) &&
+                    !outerTypes.Any(outerType => outerType.Equals(symbol.ContainingType)))
+                {
+                    return true;
+                }
 
-            return symbols;
+                // The symbols may include local variables that are declared later in the method and should not be
+                // included in the completion list, so remove those. Filter them away, unless we're in the debugger,
+                // where we show all locals in scope.
+                if (filterOutOfScopeLocals && symbol.IsInaccessibleLocal(context.Position))
+                    return true;
+
+                if (IsCapturedPrimaryConstructorParameter(context, enclosingNamedType, symbol, cancellationToken))
+                    return true;
+
+                return false;
+            }
+
+            static bool IsCapturedPrimaryConstructorParameter(
+                CSharpSyntaxContext context,
+                INamedTypeSymbol? enclosingNamedType,
+                ISymbol symbol,
+                CancellationToken cancellationToken)
+            {
+                if (enclosingNamedType is null)
+                    return false;
+
+                if (symbol is not IParameterSymbol parameterSymbol)
+                    return false;
+
+                if (!parameterSymbol.IsPrimaryConstructor(cancellationToken))
+                    return false;
+
+                // Fine to offer primary constructor parameters in field/property initializers 
+                var initializer = context.TargetToken.GetAncestors<EqualsValueClauseSyntax>().FirstOrDefault();
+                if (initializer is
+                    {
+                        Parent: PropertyDeclarationSyntax or
+                                VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax { Parent: FieldDeclarationSyntax } }
+                    })
+                {
+                    return false;
+                }
+
+                // We're not in an initializer.  Filter out this primary constructor parameter if it's already been
+                // captured by an existing field or property initializer.
+
+                var parameterName = parameterSymbol.Name;
+                foreach (var reference in enclosingNamedType.DeclaringSyntaxReferences)
+                {
+                    if (reference.GetSyntax(cancellationToken) is not TypeDeclarationSyntax typeDeclaration)
```

Comment:
When would this be true?

Reply:
Should be true. But I like being cautious :-)